uber/automated_emails.py

```diff
@@ -48,57 +50,66 @@ def send(self, x, raise_errors=True):
             if raise_errors:
                 raise
 
-    # TODO: joinedload on other tables such as shifts as well, for performance (this is WAY slower than it could be)
+    # TODO: joinedload on other tables such as shifts as well, for performance (this method is WAY slower than it could be)
     @classmethod
     def send_all(cls, raise_errors=False):
-        with Session() as session:
-            attendees, groups = session.everyone()
-            approved = {ae.subject for ae in session.query(ApprovedEmail).all()}
-            models = {Attendee: attendees, Group: groups, 'SeasonPass': session.season_passes()}
-            all_sent = {(e.model, e.fk_id, e.subject): e for e in session.query(Email).all()}
-            for rem in cls.instances.values():
-                if not rem.needs_approval or rem.subject in approved:
-                    for x in models[rem.model]:
-                        if rem.should_send(x, all_sent):
-                            rem.send(x, raise_errors=raise_errors)
```

Comment:
Move this if-statement inside this function is another example of only checking config at runtime rather than import time.